torch/csrc/profiler/nvtx_observer.cpp

```diff
@@ -32,15 +32,98 @@ struct NVTXThreadLocalState : ProfilerThreadLocalStateBase {
   }
 };
 
+std::pair<int, int> getOpIdFromInput(const c10::IValue& input_item) {
+  std::pair<int, int> producer_op_pair(-1, -1);
+  const at::Tensor& tensor = input_item.toTensor();
+  auto state_ptr = NVTXThreadLocalState::getTLS();
```

Comment:
I can move getOpIdFromInput to NVTXThreadLocalState, but I don't think it will work for updateOutputTensorTracker.  The reason is updateOutputTensorTracker is called from the 2nd arg to RecordFunctionCallback().  This arg is a function pointer that only takes RecordFunction and Ctx as args.  I would need to also pass in state_ptr to call updateOutputTensorTracker, but that would mean changing the signature to RecordFunctionCallback().